src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const mainAction = actions[i].type;
+
+    switch (mainAction) {
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          state[key] = actions[i].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        if (actions[i].keysToRemove !== undefined) {
```

Comment:
There's no need for checking if 'keysToRemove' is undefined. If it's undefined, the later for loop just won't execute.